naucse_render/course.py

```diff
@@ -0,0 +1,193 @@
+"""
+Retreive course meta-information
+
+Reads source YAML files and merges them to one JSON, with
+render info for items.
+"""
+
+from pathlib import Path
+import datetime
+import textwrap
+
+from .load import read_yaml
+from .markdown import convert_markdown
+
+
+API_VERSION = 1
+
+
+def encode_dates(value):
+    if isinstance(value, datetime.date):
+        return value.isoformat()
+    elif isinstance(value, dict):
+        return {k: encode_dates(v) for k, v in value.items()}
+    elif isinstance(value, (list, tuple)):
+        return [encode_dates(v) for v in value]
+    elif isinstance(value, (str, int, bool, type(None))):
+        return value
+    raise TypeError(value)
+
+
+# XXX: Clarify the version
+
+def get_course(course_slug: str, *, path='.', version):
+    """Get information about the course as a JSON-compatible dict"""
+
+    base_path = Path(path).resolve()
+
+    if version <= 0:
```

Comment:
What version is this? The API_VERSION above? Shouldn't it be used here?

```suggestion
    if version < API_VERSION:
```